### PR TITLE
Vault resilience

### PR DESCRIPTION
**Pull request by lablans**
This will make Beam survive connection requests to Vault if it is not ready yet

- [ ] Works for a sealed Vault
- [ ] Works for a Vault that has not been (fully) started yet

### CODE REVIEW

**Review by enola-dkfz — APPROVED** [review]
Splendid

File: broker/src/serve_pki.rs [hunk 1]
```diff
@@ -45,8 +45,10 @@ async fn get_certificate_by_serial(
     Path(serial): Path<String>
 ) -> Result<String, PkiError> {
     debug!("=> Asked for cert with serial {serial}");
-    let cert = shared::crypto::get_cert_and_client_by_serial_as_pemstr(&serial).await
-        .ok_or(PkiError::CommunicationWithVault(String::new()))?;
+    let cert = match tokio::time::timeout(std::time::Duration::new(10,0), shared::crypto::get_cert_and_client_by_serial_as_pemstr(&serial)).await {
+        Ok(certificate) => certificate.ok_or(PkiError::CommunicationWithVault(format!("Cannot retreive certificate for serial {serial}"))),
+        Err(e) => Err(PkiError::CommunicationWithVault(format!("Cannot request for certificate with serial {serial} timed out: {e}")))
```
Comment (lablans) [inline broker/src/serve_pki.rs:50]:
```suggestion
        Err(e) => Err(PkiError::CommunicationWithVault(format!("Request for certificate with serial {serial} timed out: {e}")))
```

File: broker/src/serve_pki.rs [hunk 2]
```diff
@@ -45,8 +45,10 @@ async fn get_certificate_by_serial(
     Path(serial): Path<String>
 ) -> Result<String, PkiError> {
     debug!("=> Asked for cert with serial {serial}");
-    let cert = shared::crypto::get_cert_and_client_by_serial_as_pemstr(&serial).await
-        .ok_or(PkiError::CommunicationWithVault(String::new()))?;
+    let cert = match tokio::time::timeout(std::time::Duration::new(10,0), shared::crypto::get_cert_and_client_by_serial_as_pemstr(&serial)).await {
+        Ok(certificate) => certificate.ok_or(PkiError::CommunicationWithVault(format!("Cannot retreive certificate for serial {serial}"))),
```
Comment (lablans) [inline broker/src/serve_pki.rs:49]:
```suggestion
        Ok(certificate) => certificate.ok_or(PkiError::CommunicationWithVault(format!("Cannot retrieve certificate for serial {serial}"))),
```

File: broker/src/serve_pki.rs (hunk 1 above)
Comment (lablans) [inline broker/src/serve_pki.rs:50]: Please don't forget to actually print these errors to the console (`error!()`)

**Pull request merged**